src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,348 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.function.BiFunction;
+
+/**
+ * Defines <Code>Parser</Code> class.
+ * <p>
+ *  <Code>Parser</Code> class is in charge of the following:
+ *  <ul>
+ *      <li>Parse user inputs given as `String`.</li>
+ *      <li>Check for erroneous inputs.</li>
+ *      <li>Return relevant command to handle input.</li>
+ *  </ul>
+ * </p>
+ */
+public class Parser {
+    /**
+     * <Code>BiFunction</Code> to add task to `taskList`.
+     * <p>
+     *  Accepts:
+     *  <ol>
+     *     <li><Code>TaskList</Code> object of the running <Code>Duke</Code>
+     *     program.</li>
+     *     <li>Full <Code>String</Code> input from user.</li>
+     *  </ol>
+     *  Returns:
+     *  <ol>
+     *     <li><Code>String</Code> output to be displayed to user.</li>
+     *  </ol>
+     * </p>
+     */
+    private static final BiFunction<TaskList, String, String> addTask =
+            TaskList::add;
+
+    /**
+     * <Code>BiFunction</Code> to delete task from <Code>taskList</Code>.
+     * <p>
+     *  Accepts:
+     *  <ol>
+     *     <li><Code>TaskList</Code> object of the running <Code>Duke</Code>
+     *     program.</li>
+     *     <li>Full <Code>String</Code> input from user.</li>
+     *  </ol>
+     *  Returns:
+     *  <ol>
+     *     <li><Code>String</Code> output to be displayed to user.</li>
+     *  </ol>
+     * </p>
+     */
+    private static final BiFunction<TaskList, String, String> deleteTask =
+            TaskList::delete;
+
+    /**
+     * <Code>BiFunction</Code> to mark task as done.
+     * <p>
+     *  Accepts:
+     *  <ol>
+     *     <li><Code>TaskList</Code> object of the running <Code>Duke</Code>
+     *     program.</li>
+     *     <li>Full <Code>String</Code> input from user.</li>
+     *  </ol>
+     *  Returns:
+     *  <ol>
+     *     <li><Code>String</Code> output to be displayed to user.</li>
+     *  </ol>
+     * </p>
+     */
+    private static final BiFunction<TaskList, String, String> mark = (
```

Comment:
I don't think this is a big issue. Perhaps you can change the method naming to markTask, since you are using names like addTask and deleteTask too?

Reply:
Thank you for pointing that out! I have rectified this in [commit f6a37d2](https://github.com/nus-cs2103-AY2223S1/ip/pull/15/commits/f6a37d2746efaaa72e4406874f531579b2db729b).